python_modules/dagster/dagster/_core/definitions/reconstruct.py

```diff
@@ -668,13 +710,20 @@ def repository_def_from_target_def(target: object) -> Optional["RepositoryDefini
         )
     elif isinstance(target, RepositoryDefinition):
         return target
+    elif isinstance(target, PendingRepositoryDefinition):
+        # must regenerate metadata
+        if repository_load_data is None:
+            repository_load_data = target.get_repository_load_data()
```

Comment:
to get something different in the stack trace on cold load we need a different method to hit on `target` here that internally does the load -> resolve